### PR TITLE
type inferred from the default value

### PR DESCRIPTION
**Pull request by Mouvedia**

### CODE REVIEW

**Comment by tomek-he-him** [comment]
:+1:

@Mouvedia @ericelliott when we give lots of flexibility it may be good to suggest a best practice. For example, the awesomely fantastic http://elmlang.org compiler infers types with 100% accuracy – but it recommends describing types explicitly for the public API.

There are [good reasons](https://github.com/eeue56/elm-for-web-developers/tree/742ab4f4f1e67ada96da726d4207cb6b24ff9268#why-does-start-app-give-me-an-unexpected-type-error) for this.

**Comment by Mouvedia** [comment]
IMHO accepting union types shouldn't be the norm hence the main use case is to only accept one type. And I love flexibility.

File: README.md [hunk 1]
```diff
@@ -69,9 +69,14 @@ Function types are described by a **function signature**. The function signature
 To make the signature familiar to readers, we use common JavaScript idioms such as destructuring, defaults, and rest parameters:
 
 ```js
-({ count = 0: Number }) => Any
 (...args: String[]) => Any
 (myArray[]) => Any
+({ count = 0: Number }) => Any
+```
+
+If the type of the parameter is not an [union](#union-types), you can omit the type if it has a default value. Taking the previous signature as an example the type is derivable from the default value:
```
Comment (ericelliott) [inline README.md]: Try:

If a parameter has a default value, most built-in types can be inferred:

``` js
({ count = 0 }) => Any
```

If the type is too complex to infer from the default value, you should still specify the type:

``` js
({ collection = []: Array | Object }) => Any
```

**Comment by tomek-he-him** [comment]
@Mouvedia, what’s called union types in Elm is completely different than what we call union types. An Elm union type is comparable to a traditional enum, but [much more powerful](http://elm-lang.org/guide/model-the-problem). They are also extremely common in Elm apps.

But the practice of annotating top-level functions in Elm has no more to do with union types than with any other type. It’s just of great help when debugging – which my previous link explains.

**Comment by Mouvedia** [comment]
I dunno if it's the right place to chat. Let's have an official IRC `#rtype` channel on Freenode?

**Comment by ericelliott** [comment]
Feel free to chat on [JavaScript Questions](https://gitter.im/learn-javascript-courses/javascript-questions). I'm frequently lurking there.

**Comment by tomek-he-him** [comment]
:+1:

Fair enough. Having slept over this, I don’t think there’s room for improvement here! :+1: :+1:

**Pull request merged**